common/foreign.yaml

```diff
@@ -0,0 +1,26 @@
+datasets:
+- data_type_def: ForeignFieldTable
+  doc: A table for identifying which objects in a file exist
+       outside the file
+  dtype:
+   - name: id
+     dtype: uint
+     doc: the unique identifier in this table
+   - name: data_type
+     dtype: text
+     doc: the data_type of the foreign object
+   - name: namespace
+     dtype: text
+     doc: the namespace that data_type belongs to
+   - name: parent_object_id
+     dtype: text
+     doc: the UUID for the parent object that contains the foreign object
+   - name: object_id
+     dtype: text
+     doc: the UUID for the object that is foreign
+   - name: path
+     dtype: text
+     doc: the location of the object in the file
```

Comment:
An explicit format for this path should be specified, i.e., the path should use backslash as a separator before a nested group or dataset, should not start with a backslash, and should be relative to the file root. For example, if the object is named "nested_group" which is inside "parent_group" which is at the root level of the file, then the path to the object is "parent_group/nested_group".

Reply:
We could create a dtype for this, similar to how we have an isodatetime string